### ISSUE TITLE
Await interceptors

### ISSUE THREAD

**Issue by valerauko** [htihospitality/re-dash#13]
The README suggests coeffects as a solution for loading from local storage. However local storage (shared_preferences) is async, so any coeffect calculation in inject-cofx that involves an await won't actually be reflected in the coeffects passed to the event.

Is there already a way to work around this or does it have to be considered at the library level?

**Comment by wkok** [comment]
For shared_preferences there is a solution provided for in [samples](https://github.com/htihospitality/re-dash/tree/main/samples/local_storage) which demonstrates holding a reference to the shared_preferences instance in an atom. This approach allows for an `init` step from `main` on which one time settings can be configured on the shared_preferences instance, for example the storage prefix.

However, this issue is still valid in other situations where a coeffect returns a future, so this has been addressed in branch [future-coeffects](https://github.com/htihospitality/re-dash/tree/future-coeffects)

Accompanying [test / example here](https://github.com/htihospitality/re-dash/blob/future-coeffects/test/hti/re_dash_test.cljd#L259)

This has one drawback in terms of supporting re-frame's `dispatch-sync`  which does not work anymore due to `await` propagating up in the call stack. I do not think this is a show stopper because:

- `dispatch-sync` was never intended to be the default way of dispatching events anyway (useful in tests though)
- there is a way to achieve the same result as `dispatch-sync`  with `(await (rd/dispatch [:event-id]))`

accordingly, I've deprecated `dispatch-sync`
